### PR TITLE
managedClasspath fails to have libraryDependency with classifiers 

### PR DESCRIPTION
**Pull request by trentontrees**
This is related to issue #461 and #444

I've gone thru and created a small test case which hopefully uncovers the error I'm seeing **or** helps me in understanding what I'm dong wrong in my sbt file.

When running the test via ```$ sbt ++2.10.6 "sbt-coursier/scripted sbt-coursier/classifiers"``` I was getting errors with the downloading from the https://repo2.maven.org.

To reproduce the issue:
```
$ cd sbt-coursier/src/sbt-test/sbt-coursier/classifiers
$ sbt -Dplugin.version=1.0.0-RC6 run
```

To see it pass:
```
$ cd sbt-coursier/src/sbt-test/sbt-coursier/classifiers
$ sbt -Dplugin.version=1.0.0-M15 run
```

### CODE REVIEW

File: sbt-coursier/src/sbt-test/sbt-coursier/classifiers/build.sbt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+scalaVersion := "2.11.8"
+
+libraryDependencies += ("org.jclouds.api" % "nova" % "1.5.9")
+  .classifier("tests")
+
+classpathTypes += "test-jars"
+
+resolvers ++= Seq(
+  MavenRepository("Maven Central", "http://central.maven.org/maven2/")
```
Comment (alexarchambault) [inline sbt-coursier/src/sbt-test/sbt-coursier/classifiers/build.sbt]: Adding that one shouldn't be necessary (it's already added by default).

Comment (trentontrees) [inline sbt-coursier/src/sbt-test/sbt-coursier/classifiers/build.sbt]: I switched repos to ```MavenRepository("Repo1", Resolver.DefaultMavenRepositoryRoot)```

that made no difference.  I'm not seeing the artifact when I do ```show managedClasspath```

Comment (dwijnand) [inline sbt-coursier/src/sbt-test/sbt-coursier/classifiers/build.sbt]: From Last-Modified, ETag, Fastly-Debug-Digest, etc they look to be the same thing.

```bash
09:37:23 $ curl -I http://central.maven.org/maven2/
HTTP/1.1 200 OK
Last-Modified: Thu, 25 May 2017 14:44:47 GMT
ETag: "0d09a809c6c2773b92e1dfe48509cb38"
Content-Type: text/html
Via: 1.1 varnish
Fastly-Debug-Digest: 9648f889e542c791069a3712497fbde4749f85854bc0f9a7409e5b5c0130957f
Content-Length: 110238
Accept-Ranges: bytes
Date: Tue, 04 Jul 2017 08:37:35 GMT
Via: 1.1 varnish
Age: 3433857
Connection: keep-alive
X-Served-By: cache-iad2150-IAD, cache-lhr6333-LHR
X-Cache: MISS, HIT
X-Cache-Hits: 0, 4
X-Timer: S1499157456.894148,VS0,VE0


09:37:36 $ curl -I http://repo1.maven.org/maven2/
HTTP/1.1 200 OK
Last-Modified: Thu, 25 May 2017 14:44:47 GMT
ETag: "0d09a809c6c2773b92e1dfe48509cb38"
Content-Type: text/html
Via: 1.1 varnish
Fastly-Debug-Digest: 9648f889e542c791069a3712497fbde4749f85854bc0f9a7409e5b5c0130957f
Content-Length: 110238
Accept-Ranges: bytes
Date: Tue, 04 Jul 2017 08:37:40 GMT
Via: 1.1 varnish
Age: 3433861
Connection: keep-alive
X-Served-By: cache-iad2150-IAD, cache-lhr6330-LHR
X-Cache: MISS, HIT
X-Cache-Hits: 0, 6
X-Timer: S1499157460.454570,VS0,VE0
```

**Comment by alexarchambault** [comment]
@trentontrees I included that in https://github.com/coursier/coursier/pull/612, which makes the manual `classpathTypes += ...` not required anymore. The test passes now.

It was already working fine in `1.0.0-RC6` btw, with `classpathTypes += "test-jar"` (instead of `"tests-jars"`).

**Pull request closed without merging**